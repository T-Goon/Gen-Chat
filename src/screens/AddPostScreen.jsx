import React, { useContext, useState } from "react";
import { View, StyleSheet } from "react-native";
import jwt_decode from "jwt-decode";

import { Context } from "../Context";

import FormInput from "../components/FormInput";
import ButtonPrim from "../components/ButtonPrim";

const AddPostScreen = ({ navigation }) => {
    const [message, setMessage] = useState('');
    const { sendMessage, getValueFor } = useContext(Context);

    const post = async () => {
        console.log(`posting: ${message}`)
        if (message !== '') {
            const token = await getValueFor('token');
            if (!token) {
                console.log('not Logged in');
                alert('Not logged in.');
                navigation.reset({ index: 1, routes: [{ name: 'Login' }] });
            } else {
                console.log('sending');
                const decoded = jwt_decode(token);
                sendMessage(JSON.stringify({
                    username: decoded.username,
                    message
                }));
            }

        }

    };

    return (
        <View style={styles.container}>
            <View style={styles.form}>
                <FormInput label='Message:' labelStyles={styles.labels} multiline={true} placeholder='message'
                    onChangeText={setMessage} />
            </View>

            <ButtonPrim buttonStyles={styles.postButton} onPress={post} >Post</ButtonPrim>
        </View>
    );
};

const styles = StyleSheet.create({
    container: {
        flex: 1,
        padding: 15,
        alignItems: 'center',
        backgroundColor: 'white'
    },
    form: {
        width: '100%',
        paddingTop: 10,
        paddingBottom: 20,
        paddingHorizontal: 15,
        borderRadius: 20,
        borderColor: 'black',
        borderWidth: 3,
        backgroundColor: 'hsl(230, 100%, 80%)',
        marginBottom: 20
    },
    labels: {
        fontSize: 20,
        fontWeight: 'bold'
    },
    postButton: {
        alignSelf: 'center'
    }
});

export default AddPostScreen;