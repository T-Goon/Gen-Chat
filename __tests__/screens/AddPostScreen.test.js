import React from 'react';
import { render } from '@testing-library/react-native';

import AddPostScreen from '../../src/screens/AddPostScreen';

describe('<AddPostScreen/>', () => {
    beforeEach(async () => {
        global.WebSocket = WebSocket;
    });

    it('has i child', async () => {
        const tree = render(<AddPostScreen />).toJSON();
        // console.log(tree.children[0].children[0].children[1]);
        expect(tree.children.length).toBe(2);
    });

    // it('can send message', async () => {

    //     const { findByText, findByPlaceholderText } = render(
    //         <ContextProvider>
    //             <AddPostScreen />
    //         </ContextProvider>);



    //     SecureStore.getItemAsync.mockResolvedValue("token");

    //     await server.connected;
    //     console.log(await server.connected);



    //     fireEvent(await findByPlaceholderText('message'), 'onChangeText', 'hello');


    //     fireEvent(await findByText('Post'), 'onPress');

    //     console.log(server);

    //     // client.send('hello');

    //     // console.log(tree);
    //     // let tree;
    //     // act(() => {
    //     //     tree = renderer.create(
    //     //         <ContextProvider>
    //     //             <AddPostScreen />
    //     //         </ContextProvider>).toJSON();
    //     //     console.log(tree);
    //     //     console.log(tree.children[0].children[0].children[1]);
    //     //     tree.children[0].children[0].children[1].props.onChangeText('hello');
    //     // tree.children[1].props.onClick();


    //     // const screen = shallow(
    //     //     <ContextProvider>
    //     //         <AddPostScreen />
    //     //     </ContextProvider>);
    //     // console.log(screen.find("*"));
    //     // screen.find(ButtonPrim).simulate('click');
    //     // screen.find(ButtonPrim).simulate('textChange', 'hello');

    //     // });

    //     await expect(server).toReceiveMessage("hello");
    //     expect(server).toHaveReceivedMessages(["hello"]);
    // });

    it('renders correctly', async () => {
        const tree = render(<AddPostScreen />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });


});