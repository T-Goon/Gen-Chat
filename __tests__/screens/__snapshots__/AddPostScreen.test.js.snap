// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddPostScreen/> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "padding": 15,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "hsl(230, 100%, 80%)",
        "borderColor": "black",
        "borderRadius": 20,
        "borderWidth": 3,
        "marginBottom": 20,
        "paddingBottom": 20,
        "paddingHorizontal": 15,
        "paddingTop": 10,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "column",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        Message:
      </Text>
      <TextInput
        allowFontScaling={true}
        keyboardType="default"
        multiline={true}
        onChangeText={[Function]}
        placeholder="message"
        rejectResponderTermination={true}
        secureTextEntry={false}
        style={
          Object {
            "backgroundColor": "hsl(0, 0%, 95%)",
            "borderColor": "black",
            "borderRadius": 5,
            "borderWidth": 2,
            "paddingHorizontal": 10,
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "hsl(550, 100%, 50%)",
        "borderRadius": 10,
        "justifyContent": "center",
        "opacity": 1,
        "paddingHorizontal": 16,
        "paddingVertical": 8,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "bold",
        }
      }
    >
      Post
    </Text>
  </View>
</View>
`;
